engine/runtime/src/main/java/org/enso/interpreter/runtime/ModuleSources.java

```diff
@@ -0,0 +1,50 @@
+package org.enso.interpreter.runtime;
+
+import com.oracle.truffle.api.TruffleFile;
+import com.oracle.truffle.api.source.Source;
+import java.io.IOException;
+import org.enso.pkg.QualifiedName;
+import org.enso.polyglot.LanguageInfo;
+import org.enso.text.buffer.Rope;
+
+
+/** Keeps information about various kinds of sources associated with a {@link Module}.
+ */
+record ModuleSources(
+          TruffleFile sourceFile,
+          Rope ropeHolder,
+          Source cachedSource) {
+
+  static final ModuleSources NONE = new ModuleSources(null, null, null);
+
+  ModuleSources newWith(TruffleFile f) {
+    return new ModuleSources(f, ropeHolder(), cachedSource());
```

Comment:
Shouldn't the 2nd and 3rd value be `null`?

Reply:
Anything breaks with that?

Reply:
Changed to `null` in [2d9e4ef](https://github.com/enso-org/enso/pull/3508/commits/2d9e4efd1cd73fade21baed4e4dcfa7945eeb436)